docs/dev/testing.rst

```diff
@@ -0,0 +1,263 @@
+
+
+Running Other Commands with Tox
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+You can run any of the individual test environments that ``tox -av`` lists on
+their own.  Not all of them literally run tests. For instance, to lint and
+build the documentation you can run:
+
+.. code-block:: console
+
+    $ tox -e docs
+
+To run all of the code and documentation linters, but not run any of the other
+tests:
+
+.. code-block:: console
+
+    $ tox -e linters
+
+Each of the test environments defined in ``tox.ini`` is just a collection of
+dependencies and commands. To see what they consist of, you can open the file
+in your text editor. Each section starts with ``[testenv:xxxxxx]`` and the
+section called ``commands`` is a list of shell commands that that test
+environment will run.
+
+-------------------------------------------------------------------------------
+Input Data for Integration Tests
+-------------------------------------------------------------------------------
+The software integration tests need a year's worth of input data to process. By
+default they will look in your local PUDL datastore to find it. If the data
+they need isn't available locally, they will download it from Zenodo and put it
+in the local datastore.
+
+However, if you're editing code that affects how the datastore works, you
+probably don't want to risk contaminating your working datastore. You can
+use a disposable temporary datastore instead by having Tox pass the
+``--tmp-data`` flag in to ``pytest`` like this:
+
+.. code-block:: console
+
+   $ tox -e integration -- --tmp-data
+
+The floating ``--`` isn't a typo, it tells Tox that you're done giving it
+command line arguments, and that any additional arguments it gets should be
+passed through to ``pytest``. We've configured ``pytest`` (through the
+``test/conftest.py`` configuration file) to be on the lookout for the
+``--tmp-data`` flag and act accordingly.
+
+.. seealso::
+
+    * :doc:`/usage` for more on how to set up a PUDL workspace, including a
+      datastore.
+    * :doc:`/datastore` for more on how to work with the datastore.
+
+-------------------------------------------------------------------------------
+Running pytest Directly
+-------------------------------------------------------------------------------
+Running tests directly with ``pytest`` gives you the ability to run only
+tests from a particular test module, or even a single individual test case.
+It's also faster because there's no testing environment to set up. Instead,
+it just uses your Python environment, which should be the ``pudl-dev`` conda
+environment discussed in :doc:`/dev/dev_setup`. This is convenient if you're
+debugging something specific, or developing new test cases, but it's not as
+robust as using Tox.
+
+Running specific tests
+^^^^^^^^^^^^^^^^^^^^^^
+To run the software unit tests with ``pytest`` directly:
+
+.. code-block:: console
+
+   $ pytest test/unit
```

Comment:
Mention that this is the same as `tox -e unit`

Reply:
okay